### PR TITLE
Reuse `:help` and `:play` frames by building frame stacks from internal link clicks

### PR DESCRIPTION
**Pull request by oskarhane**
The stream becomes cluttered quite quickly when clicking on help topics from within help topics.
This PR enables frame stacks, so if a user clicks a help topic from within a help topic, the same frame is reused and the user can navigate back and forth through out the stack.

The same goes for guides.

Frames are only reused within their own category, so clicking a guide topic from within a help topic will generate a new frame in the stream.

Demo guides:  
Mar-24-2020 18-49-30

Demo help:
Mar-24-2020 19-26-46

### CODE REVIEW

File: src/browser/modules/Stream/HelpFrame.jsx [hunk 1]
```diff
@@ -17,27 +17,86 @@
...
+  useEffect(() => {
+    setCurrentFrameIndex(0)
+  }, [stack.length])
+
+  const { aside, main } = generateContent(currentFrame)
+
+  const prevBtn =
+    currentFrameIndex === stack.length - 1 ? null : (
+      <CarouselButton
+        className="previous-slide  rounded"
```
Comment (HerrEmil) [inline src/browser/modules/Stream/HelpFrame.jsx]: double space

File: src/browser/modules/Stream/PlayFrame.jsx [hunk 2]
```diff
@@ -43,145 +50,216 @@ const checkHtmlForSlides = html => {
...
+  }
+
+  useEffect(() => {
+    setStackIndex(0)
+  }, [stack.length])
+
+  const prevBtn =
+    stackIndex === stack.length - 1 || !atSlideStart ? null : (
+      <CarouselButton
+        className="previous-slide  rounded"
```
Comment (HerrEmil) [inline src/browser/modules/Stream/PlayFrame.jsx]: double space

File: src/browser/modules/Stream/PlayFrame.jsx [hunk 3]
```diff
@@ -43,145 +50,216 @@ const checkHtmlForSlides = html => {
...
+  const guide = chapters[guideName] || {}
+  // Check if content exists locally
+  if (Object.keys(guide).length) {
+    const { content, title, subtitle, slides } = guide
+    return {
+      guide: (
+        <Docs
+          originFrameId={stackFrame.id}
+          withDirectives
+          content={!!slides ? null : content}
```
Comment (HerrEmil) [inline src/browser/modules/Stream/PlayFrame.jsx]: I don't think the double not operator changes what the ternary evaluates to

File: src/browser/modules/Stream/stream.utils.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+export function getLatestFromFrameStack(frameObj) {
+  if (!frameObj.stack) {
```
Comment (HerrEmil) [inline src/browser/modules/Stream/stream.utils.js]: Could be `frameObj => frameObj.stack && frameObj.stack[0] || null`

Comment (oskarhane) [reply src/browser/modules/Stream/stream.utils.js]: I agree, but I think it makes it a little bit harder to read tbh.

**Pull request merged**